insights/parsers/dnf_module.py

```diff
@@ -17,42 +17,90 @@
 from insights.specs import Specs
 
 
+class Profile(object):
+    """
+    An object for the dnf module:stream profile
+
+    Attributes:
+        profile (Optional[str]): Profile of the dnf module:stream
+        default (bool): Default flag of the dnf module:stream profile
+        installed (bool): Installed flag of the dnf module:stream profile
+    """
+    def __init__(self, profile):
+        self.profile = profile.split(" ")[0]
+        self.default = "[d]" in profile
+        self.installed = "[i]" in profile
+
+
+class DnfModuleStream(object):
+    """
+    An object for the dnf module:stream
+
+    Attributes:
+        stream (str): Name of the module stream
+        profiles (list): List of profiles of the dnf module:stream
+        summary (str): Summary of the dnf module:stream
+        default (bool): Default flag of the dnf module:stream
+        enabled (bool): Enabled flag of the dnf module:stream
+        disabled (bool): Disabled flag of the dnf module:stream
+        installed (bool): Installed flag of the dnf module:stream
+        active (Optional[bool]): Active flag of the dnf module:stream
+    """
+    def __init__(self, data=None, cmd="list"):
+        data = data or {}
+        self._cmd = cmd
+        self._stream = data.get('Stream', '')
+        self.stream = self._stream.split(" ")[0]
+        self.profiles = [Profile(p.strip()) for p in data.get('Profiles', '').split(',') if p.strip()]
+        self.summary = data.get('Summary', '')
+        self.default = "[d]" in self._stream
+        self.enabled = "[e]" in self._stream
+        self.disabled = "[x]" in self._stream
+        self.installed = "[i]" in self._stream
+        self.active = self.enabled
+        if self._cmd == "info":
+            # only `dnf module info` shows `[a]` flag
+            self.active = "[a]" in self._stream
+
+
 class DnfModuleBrief(object):
     """
     An object for the dnf modules listed by ``dnf module list`` command
 
     Attributes:
         name (str): Name of the dnf module
-        stream (str): Stream of the dnf module
-        profiles (list): List of profiles of the dnf module
-        summary (str): Summary of the dnf module
+        streams (list): List of streams of the dnf module
```

Comment:
There might be a compatibility issue, when changing from `stream` to `streams`.

Reply:
it is a breaking change which is not compatible at all, previously it was `stream` string, now it is a list

Reply:
TBH, this PR changes the Interface of the existing parser, that's to say, it would break the rules/services that depend on it. However, as you mentioned, this spec was removed from the collection and was not collected by Insights for a long time, and it fixes some issues which were incorrect. It looks good to me